Update copyright to cert-manager project

meyskens opened the pull request:
**What this PR does / why we need it**:
This updates the copyright to  The cert-manager contributors` to follow the CNCF donation and pending repo move outside of jetstack

**Release note**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access)
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`.
-->
```release-note
Change copyright owner to `The cert-manager Authors`
```

meyskens commented:
@JoshVanL i'd like to propose the award for touching most area's in the codebase in 1 PR :joy:

meyskens commented:
/kind cleanup

wallrj reviewed (commented):
Hey @meyskens 

A couple of things:

 * I thought when we discussed this the other day, in slack, we concluded that it should be "The cert-manager Authors." in keeping with the [Kubernetes boilerplate](https://github.com/kubernetes/kubernetes/blob/release-1.20/hack/boilerplate/boilerplate.go.txt) and because we distinguish between authors of code and contributors such as issue reporters etc. 

* I'd also want to double check the significance of the "Copyright YEAR". My understanding is that the the year should match the year that the file came into existence so it's clear that the copyright covers the whole history of the file...

wallrj reviewed (commented):
Another thing I notice is that there are some non Jetstatck LICENCE files in the repo:

```
find . -type f -iname license 
./logo/LICENSE
./pkg/issuer/acme/dns/util/LICENSE
./pkg/issuer/acme/dns/cloudflare/LICENSE
./pkg/issuer/acme/dns/azuredns/LICENSE
./pkg/issuer/acme/dns/route53/LICENSE
./pkg/issuer/acme/dns/clouddns/LICENSE
```

With different copyright names. What do we do with those?

meyskens commented:
> The cert-manager Authors

Updated that!

> I'd also want to double check the significance of the "Copyright YEAR"

We did change copyrights in the past, i have updated it all now as the copyright owner changed. The historic copyright is already enforced by it having a licence in the Git history

> With different copyright names. What do we do with those?

Well be good to have some legal people look into the logo, others seem to be those because we 1:1 copied then which then the licence would be fair to keep

wallrj reviewed (approved):
/lgtm 

xref: https://github.com/jetstack/cert-manager/issues/3451 https://github.com/cncf/toc/issues/557

munnerz commented:
> We did change copyrights in the past, i have updated it all now as the copyright owner changed. The historic copyright is already enforced by it having a licence in the Git history

Not too sure if this is correct 🤔 it also means anyone wanting to review the licenses we make use of would have to trawl every revision to be safe. Would it make sense to keep these as is, or alternatively move them to a `third_party` directory?
